frontend/pages/hosts/details/DeviceUserPage/DeviceUserPage.tsx

```diff
@@ -63,11 +63,7 @@ const DeviceUserPage = ({
       refetchOnReconnect: false,
       refetchOnWindowFocus: false,
       retry: false,
-      select: (data: IDeviceMappingResponse) =>
-        data.device_mapping &&
-        data.device_mapping.filter(
-          (deviceUser) => deviceUser.email && deviceUser.email.length
-        ),
+      select: (data: IDeviceMappingResponse) => data.device_mapping,
```

Comment:
This change will return the entire `device_mapping` array without filtering out devices with no email. Are you moving that check to the backend?

Reply:
Yes! We're doing two things:

1. Ensure we don't ingest empty emails anymore
2. Cleaning empty emails from the db